### PR TITLE
More windows hardcode

### PR DESCRIPTION
**Pull request by TheGoatGod**
#### Summary

SUMMARY: Infrastructure "fixes more windows added a while back, you can now tear curtains and adds new code to base game"

#### Purpose of change

allow more windows to be added to the game with ease

#### Describe the solution

add new code to allow "curtain_transform": "t_window_no_curtains", 
now you don't have to add hardcode to dismantle curtains

#### Describe alternatives you've considered

adding heaps of code to src

#### Testing

loaded up and everything works as intended

### CODE REVIEW

**Comment by TheGoatGod** [comment]
can someone help with the checks i don't have time rn but I will go through them when I get time

File: src/iexamine.cpp [hunk 1]
```diff
@@ -4003,14 +4003,8 @@ void iexamine::curtains( player &p, const tripoint &examp )
...
-        if( ter == t_window_domestic || ter == t_curtains ) {
-            here.ter_set( examp, t_window_no_curtains );
-        } else if( ter == t_window_open ) {
-            here.ter_set( examp, t_window_no_curtains_open );
-        } else if( ter == t_window_domestic_taped ) {
-            here.ter_set( examp, t_window_no_curtains_taped );
-        } else if( ter == t_window_bars_domestic || ter == t_window_bars_curtains ) {
-            here.ter_set( examp, t_window_bars );
+        if (here.ter(examp)->has_curtains()) {
+            here.ter_set(examp, here.ter(examp)->curtain_transform();
```
Comment (AMurkin) [inline src/iexamine.cpp:4007]: Probably this. But you must compile your code.
```suggestion
            here.ter_set(examp, here.ter(examp)->curtain_transform;
```

File: src/mapdata.h [hunk 2]
```diff
@@ -247,6 +250,14 @@ struct map_data_common_t {
         std::set<std::string> flags;    // string flags which possibly refer to what's documented above.
         std::bitset<NUM_TERFLAGS> bitflags; // bitfield of -certain- string flags which are heavily checked
 
+    public:
+        ter_str_id curtain_transform;
+
+    bool has_curtains() const
+    {
+        return !(curtain_transform.is_empty() || curtain_transform.is_null());
+    }
```
Comment (AMurkin) [inline src/mapdata.h:259]:
```suggestion
    bool has_curtains() const
        bool has_curtains() const {
            return !( curtain_transform.is_empty() || curtain_transform.is_null() );
        }
```

File: src/mapdata.cpp [hunk 3]
```diff
@@ -1161,6 +1161,7 @@ void map_data_common_t::load( const JsonObject &jo, const std::string &src )
     }
 
     mandatory( jo, was_loaded, "description", description );
+    optional( jo, was_loaded, "curtain_transform", curtain_transform);
```
Comment (AMurkin) [inline src/mapdata.cpp:1164]:
```suggestion
    optional( jo, was_loaded, "curtain_transform", curtain_transform );
```

File: src/iexamine.cpp [hunk 4]
```diff
@@ -4003,14 +4003,8 @@ void iexamine::curtains( player &p, const tripoint &examp )
...
-        if( ter == t_window_domestic || ter == t_curtains ) {
-            here.ter_set( examp, t_window_no_curtains );
-        } else if( ter == t_window_open ) {
-            here.ter_set( examp, t_window_no_curtains_open );
-        } else if( ter == t_window_domestic_taped ) {
-            here.ter_set( examp, t_window_no_curtains_taped );
-        } else if( ter == t_window_bars_domestic || ter == t_window_bars_curtains ) {
-            here.ter_set( examp, t_window_bars );
+        if (here.ter(examp)->has_curtains()) {
+            here.ter_set(examp, here.ter(examp)->curtain_transform;
```
Comment (AMurkin) [inline src/iexamine.cpp:4007]:
```suggestion
        if( here.ter( examp )->has_curtains() ) {
            here.ter_set( examp, here.ter( examp )->curtain_transform );
```

File: src/mapdata.h [hunk 5]
```diff
@@ -247,6 +250,14 @@ struct map_data_common_t {
         std::set<std::string> flags;    // string flags which possibly refer to what's documented above.
         std::bitset<NUM_TERFLAGS> bitflags; // bitfield of -certain- string flags which are heavily checked
 
+    public:
+        ter_str_id curtain_transform;
+
+    bool has_curtains() const
```
Comment (AMurkin) [inline src/mapdata.h:256]:
```suggestion
```

**Comment by TheGoatGod** [comment]
not sure what needs done to fix this? i can in the meantime fix the descriptions when I get time

**Comment by anothersimulacrum** [comment]
At a glance, this looks good to go.

**Pull request merged**